packages/hash/api/src/graphql/typeDefs/knowledge/entity.typedef.ts

```diff
@@ -1,87 +1,37 @@
 import { gql } from "apollo-server-express";
 
-export const persistedEntityTypedef = gql`
-  interface PersistedEntity {
-    # These fields are repeated everywhere they're used because
-    # (a) GQL requires it - https://github.com/graphql/graphql-spec/issues/533
-    # (b) string interpolation breaks the code generator's introspection
-    #
-    # Could maybe use a custom schema loader to parse it ourselves:
-    # https://www.graphql-code-generator.com/docs/getting-started/schema-field#custom-schema-loader
-    #
-    # For now, _COPY ANY CHANGES_ from here to any type that 'implements Entity'
-    """
-    The id of the entity
-    """
-    entityId: ID!
-    """
-    The specific version of the entity
-    """
-    entityVersion: String!
-    """
-    The id of the account that owns this entity.
-    """
-    ownedById: ID!
-    """
-    Alias of ownedById - the id of the account that owns this entity.
-    """
-    accountId: ID!
-      @deprecated(reason: "accountId is deprecated. Use ownedById instead.")
+export const entityWithMetadataTypedef = gql`
+  scalar EntityId
+  scalar EntityEditionId
+  scalar PropertyObject
+  scalar EntityMetadata
+
+  interface EntityWithMetadata {
     """
-    The versioned URI of this entity's type.
+    Metadata for the entity.
     """
-    entityTypeId: String!
+    metadata: EntityMetadata!
     """
-    The JSON object containing the entity's properties.
+    Properties of entity.
     """
-    properties: JSONObject!
+    properties: PropertyObject!
   }
 
-  type UnknownPersistedEntity implements PersistedEntity {
-    # ENTITY INTERFACE FIELDS BEGIN #
-    """
-    The id of the entity
-    """
-    entityId: ID!
-    """
-    The specific version of the entity
+  type UnknownEntityWithMetadata implements EntityWithMetadata {
     """
-    entityVersion: String!
+    Metadata for the entity.
     """
-    The id of the account that owns this entity.
-    """
-    ownedById: ID!
-    """
-    Alias of ownedById - the id of the account that owns this entity.
-    """
-    accountId: ID!
-      @deprecated(reason: "accountId is deprecated. Use ownedById instead.")
-    """
-    The versioned URI of this entity's type.
-    """
-    entityTypeId: String!
-    """
-    The JSON object containing the entity's properties.
-    """
-    properties: JSONObject!
-    # ENTITY INTERFACE FIELDS END #
-  }
-
-  """
-  For referring to an existing entity owned by a specific account id
-  """
-  input PersistedExistingEntity {
+    metadata: EntityMetadata!
     """
-    This may be a reference to a placeholder set using placeholderId on a previous UpdatePageContentsAction.
+    Properties of entity.
     """
-    entityId: ID!
-    ownedById: ID!
+    properties: PropertyObject!
   }
 
   """
   Select entity types by ONE of componentId, entityTypeId
```

Comment:
What does this even do? Do we still need it, the docs and such don't make sense to me

Reply:
Would appreciate you looking into that, thanks

Reply:
I think I am able to get rid of this `entityTypeChoice` entirely, see 655d3221a266fdfd72cbc0e6cf6d2fd498022a42.
Page saving is only using the `entityTypeId` which is the only option anyway